Template and VF Components/QueryComponentController.cls

```diff
@@ -0,0 +1,89 @@
+public class QueryComponentController {
+    public String inputQuery {get; set;}
+    public List<String> fieldList {get; set;}
+    private Integer pageSize = 10;
+    
+    private final String SORT_DESC = 'DESC';
+    private final String SORT_ASC = 'ASC';
+    private String sortDirection;
+    public String newquery = '';
+    public String sortBy{get;set;}
+    
+    public  QueryComponentController(){
+        fieldList = new List<String> ();
+        sortDirection = SORT_DESC;
+    }
+    
+    
+    
+    public ApexPages.StandardSetController setController {
+        get {
+            if(setController==null){
+                List<SObject> records = getQueryResult();
+                setController = new ApexPages.StandardSetController(records);
+                setController.setPageSize(pageSize);
+                if(records.size() > 0)
+            	{
+            		initializeFieldList(records[0]);
+        		}
+            }
+            return setController;
+        }
+        set;
+    }
+   
+    
+    public List<SObject> getQueryResult()
+    {
+        List<SObject> result = new List<SObject>();
+        try{
+            
+            String query = createQuery(inputQuery);
+            System.debug('inside getQueryResult ---' + query);
```

Comment:
no system debug should be present when you submit the assignment. unless some debugging is still needed

Reply:
I removed it.